src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -804,6 +808,41 @@ internal async void ExecutePackageDownload(string name, PackageVersion package,
                     if (res == MessageBoxResult.Cancel || res == MessageBoxResult.None) return;
                 }
 
+                // determinate if any of the packages are targeting other hosts
+                var containsPackagesThatTargetOtherHosts = false;
+
+                // Known hosts
+                var knownHosts = Model.GetKnownHosts();
+
+                // Sandbox, special case: Warn if any package targets only one known host
+                if (String.IsNullOrEmpty(Host))
+                {
+                    containsPackagesThatTargetOtherHosts =
+                        newPackageHeaders.Any(y => y.host_dependencies != null && y.host_dependencies.Intersect(knownHosts).Count() == 1);
+                }
+                else
+                {
+                    // Warn if there are packages targeting other hosts but not our host
+                    var otherHosts = knownHosts.Except(new List<string>() {Host});
+                    containsPackagesThatTargetOtherHosts = newPackageHeaders.Any(x =>
+                    {
+                        // Is our host in the list?
+                        // If not, is any other host in the list?
+                        return x.host_dependencies != null && !x.host_dependencies.Contains(Host) && otherHosts.Any(y => x.host_dependencies != null && x.host_dependencies.Contains(y));
```

Comment:
aren't you checking for this condition twice in the same line: `x.host_dependencies != null`?